app/src/main/java/com/example/caloriecounter/feature/restaurant/ui/RestaurantScreen.kt

```diff
@@ -1,23 +1,10 @@
 package com.example.caloriecounter.feature.restaurant.ui
 
-import androidx.compose.foundation.layout.fillMaxSize
-import androidx.compose.material.Text
-import androidx.compose.runtime.Composable
-import androidx.compose.ui.Modifier
-import androidx.compose.ui.graphics.Color
-import androidx.compose.ui.text.style.TextAlign
-import androidx.compose.ui.unit.sp
-import com.example.caloriecounter.feature.restaurant.RestaurantViewModel
+sealed class RestaurantScreen(val route: String, val navToRoute: String = route) {
+    object CategoryList : RestaurantScreen("category_list")
+    data class MealList(val argument: String = "category") :
+        RestaurantScreen("meal_list/{$argument}", "meal_list/$argument")
 
-@Composable
-fun RestaurantScreen(
-    vm: RestaurantViewModel,
-) {
-    Text(
-        text = "RestaurantScreen",
-        fontSize = 24.sp,
-        color = Color.DarkGray,
-        modifier = Modifier.fillMaxSize(),
-        textAlign = TextAlign.Center,
-    )
+    data class MealRecipe(val argument: String = "mealId") :
+        RestaurantScreen("meal_recipe/{$argument}", "meal_list/$argument")
```

Comment:
It was slightly confusing for the first time to understand why do we have `route` and `navToRoute` 😄 

There is another way to interpret it if you interested (no need to rewrite anything you have for now)
1. Screen.Exercise.Opening.route `connectArgs` Screen.Exercise.ID
```kotlin
/** Use for routes args in [composable] */
infix fun String.connectArgs(args: Any): String {
  return "$this/{$args}"
}
```
2. Screen.Exercise.Opening.route `withArgs` exercise.nextId
```kotlin
/** Use to attach args to [NavHostController.navigate] */
infix fun String.withArgs(args: Any): String {
  return "$this/$args"
}
```

Reply:
Thanks, It looks much better.